### PR TITLE
Build shared library for libcpgplot.

### PR DESCRIPTION
**Pull request by demorest**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by demorest** [comment]
@conda-forge-admin, please rerender

**Comment by demorest** [comment]
@pkgw, as far as I can tell this is ready to go.  But this is also the first time I've tried this so let me know if I missed anything.

File: recipe/sharedcpg.patch [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ 	$(CCOMPL) $(CFLAGC) $(SRC)/cpg/pgbind.c -o pgbind
+ 
+-libcpgplot.a cpgplot.h: $(PG_SOURCE) pgbind 
++libcpgplot.a $(SHARED_CPG_LIB) cpgplot.h: $(PG_SOURCE) pgbind 
+ 	./pgbind $(PGBIND_FLAGS) -h -w $(PG_SOURCE)
+ 	$(CCOMPL) -c $(CFLAGC) cpg*.c
+ 	rm -f cpg*.c
+ 	ar ru libcpgplot.a cpg*.o
+ 	$(RANLIB) libcpgplot.a
++	test -n "$(SHARED_CPG_LD)" && $(SHARED_CPG_LD) cpg*.o $(SHARED_LIB) $(SHARED_LIB_LIBS)
```
Comment (pkgw) [inline recipe/sharedcpg.patch:32]: Welllll ... for a really proper Makefile style, the shared libcpgplot library should be built in a separate rule. But we're just patching something onto an old, undeveloped library, so `¯\_(ツ)_/¯`.

One point on this topic: sometimes, you need to use different object files for static and shared libraries; e.g. compiled with `-fPIC -DPIC` or not. But if we run into any problems here, we'll deal with them.

**Review by pkgw — APPROVED** [review]
In principle there might be some technical issues to worry about, but given that pgplot isn't an actively maintained library, let's go ahead and merge this and fix up any problems if they emerge.

**Comment by pkgw** [comment]
Thanks, Paul!

**Pull request merged**